sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/exporter/SpanExporter.scala

```diff
@@ -0,0 +1,119 @@
+
+/** An interface that allows different tracing services to export recorded data
+  * for sampled spans in their own format.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/sdk/#span-exporter]]
+  *
+  * @tparam F
+  *   the higher-kinded type of a polymorphic effect
+  */
+trait SpanExporter[F[_]] {
+
+  /** Called to export sampled
+    * [[org.typelevel.otel4s.sdk.trace.data.SpanData SpanData]].
+    *
+    * '''Note''': the export operations can be performed simultaneously
+    * depending on the type of span processor being used. However, the batch
+    * span exporter will ensure that only one export can occur at a time.
+    *
+    * @param span
+    *   the collection of sampled Spans to be exported
+    */
+  def exportSpans(span: List[SpanData]): F[Unit]
+
+  /** Exports the collection of sampled
+    * [[org.typelevel.otel4s.sdk.trace.data.SpanData SpanData]] that have not
+    * yet been exported.
+    *
+    * '''Note''': the export operations can be performed simultaneously
+    * depending on the type of span processor being used. However, the batch
+    * span exporter will ensure that only one export can occur at a time.
+    */
+  def flush: F[Unit]
+}
+
+object SpanExporter {
+
+  /** Creates a [[SpanExporter]] which delegates all exports to the exporters in
+    * order.
+    */
+  def of[F[_]: Applicative](
+      exporters: SpanExporter[F]*
+  ): SpanExporter[F] =
+    exporters.combineAll
+
+  /** Creates a no-op implementation of the [[SpanExporter]].
+    *
+    * All export operations are no-op.
+    */
+  def noop[F[_]: Applicative]: SpanExporter[F] =
+    new Noop
+
+  implicit def spanExporterMonoid[F[_]: Applicative]: Monoid[SpanExporter[F]] =
+    new Monoid[SpanExporter[F]] {
+      val empty: SpanExporter[F] =
+        noop[F]
+
+      def combine(x: SpanExporter[F], y: SpanExporter[F]): SpanExporter[F] =
+        (x, y) match {
+          case (that, _: Noop[F]) =>
+            that
+          case (_: Noop[F], other) =>
+            other
+          case (that: Multi[F], other: Multi[F]) =>
+            Multi(that.exporters.concatNel(other.exporters))
+          case (that: Multi[F], other) =>
+            Multi(that.exporters :+ other)
+          case (that, other: Multi[F]) =>
+            Multi(that :: other.exporters)
+          case (that, other) =>
+            Multi(NonEmptyList.of(that, other))
+        }
+    }
+
+  private final class Noop[F[_]: Applicative] extends SpanExporter[F] {
+    def exportSpans(span: List[SpanData]): F[Unit] = Applicative[F].unit
+    def flush: F[Unit] = Applicative[F].unit
+
+    override def toString: String = "SpanExporter.Noop"
+  }
+
+  private final case class Multi[F[_]: Applicative](
+      exporters: NonEmptyList[SpanExporter[F]]
+  ) extends SpanExporter[F] {
+    def exportSpans(span: List[SpanData]): F[Unit] =
+      exporters.traverse_(_.exportSpans(span))
+
+    def flush: F[Unit] =
+      exporters.traverse_(_.flush)
```

Comment:
I wonder if these should be `parTraverse`, so that exporting/flushing can happen concurrently.

Reply:
I believe we can use `parTraverse`. Java SDK does something similar. They submit all tasks and then aggregate the results: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/MultiSpanExporter.java#L40-L55